### DISCUSSION TITLE
Community compartment structure - separate-vs-combined extracellular space

### DISCUSSION THREAD

**Discussion by robert-clegg-tessella** [micom-dev/micom#200] [Q&A]
Hello, thank you for this really useful tool 😄 

I've done some digging through the paper, the codebase, the documentation, and the discussions but couldn't find an answer to my question. Please point me in the right direction if I've missed it!

I'm trying to simulate a community of a subset of species from the AGORA2 dataset within custom-defined media. I get unexpected results, where growth is lower in the medium that should have more, and vice versa. I _think_ this is because when I use my more comprehensive medium definition (i.e., where many more metabolites have defined fluxes), the flux upper limit on critical metabolites is set to some small value, whereas when I use the more restrictive medium definition these flux upper limits are left at the default (1000).

All of this has made me inspect my `sample.compartments` more closely. For each species, I have a `c_...` (cytoplasm), `p_...` (periplasm), and an `e_...` (extracellular space) compartment, plus a `m` (medium) compartment that is shared between all species. This all makes sense to me except for the `e_...` compartments: shouldn't there be a shared `e` compartment for all species? Transport reactions including the `m` compartment are all to individual `e_...` compartments, meaning that any flux of metabolites between species must currently go through the `m` compartment, which has flux limits on it due to the medium definition. If there were a shared `e` compartment with no (or very high) upper bounds on flux, this exchange of metabolites wouldn't be limited by the flux limits associated with the medium definition.

Have I fundamentally misunderstood something, or have I set up my community incorrectly? Many thanks in advance for any help you're able to offer!

**Comment by vicru93** [comment]
Hi, I have the same question. Were you able to resolve it?

**Reply by cdiener** [reply]
Sorry about that. I thought I had already replied. Everything is set up correctly. The e_ compartments are the extracellular vicinity for specific taxa which is necessary to model extracellular or membrane bound enzymes. So metabolites flow between the shared m compartment into the e_ compartments and can then be transported to the cytosol.

**Reply by vicru93** [reply]
Hi,

Thanks — I've already resolved the issue.

I would like to respectfully submit the following comment:

First, thank you for developing such a powerful tool for simulating community-level metabolic interactions.

I'm working with MAGs reconstructed via ModelSEED, and I’ve noticed that MICOM supports classical FBA and pFBA, but not any form of loopless FBA. This can be problematic when working with automatically gapfilled models in anaerobic systems.

In COBRApy, there are two commonly used strategies to reduce or eliminate loops:

1. `loopless_solution(model)` — a heuristic approach that avoids thermodynamically infeasible cycles (not MILP).
2. `add_loopless(model)` — a MILP-based approach that guarantees loop removal.


**Suggestion:**
Would it be possible to integrate an option (e.g., loopless=True) in cooperative_tradeoff() or other MICOM workflows that implements one of these approaches (ideally the heuristic one, for computational efficiency), in order to reduce infeasible loops in FBA results?

I understand that implementing MILP for entire communities may be computationally expensive, but even the inclusion of a heuristic method like loopless_solution() would be a valuable addition.

Thanks again for your great work!

Best regards,
Victor
